src/functions/ducklake_options.cpp

```diff
@@ -0,0 +1,150 @@
+     {"parquet_row_group_size", "Number of rows per row group in Parquet files"},
+     {"version", "DuckLake format version"},
+     {"created_by", "DuckLake creator information"},
+     {"data_path", "Path to data files"},
+     {"encrypted", "Whether the DuckLake is encrypted"}}};
+
+struct DuckLakeOptionsData : public TableFunctionData {
+	explicit DuckLakeOptionsData(Catalog &catalog) : catalog(catalog) {
+	}
+
+	Catalog &catalog;
+};
+
+struct DuckLakeOptionInfo {
+	string option_name;
+	string description;
+	string value;
+	string scope;
+	string scope_entry;
+};
+
+struct DuckLakeOptionsState : public GlobalTableFunctionState {
+	DuckLakeOptionsState() : offset(0) {
+	}
+
+	vector<DuckLakeOptionInfo> options;
+	idx_t offset;
+};
+
+static unique_ptr<FunctionData> DuckLakeOptionsBind(ClientContext &context, TableFunctionBindInput &input,
+                                                    vector<LogicalType> &return_types, vector<string> &names) {
+	auto &catalog = BaseMetadataFunction::GetCatalog(context, input.inputs[0]);
+
+	names.emplace_back("option_name");
+	return_types.emplace_back(LogicalType::VARCHAR);
+
+	names.emplace_back("description");
+	return_types.emplace_back(LogicalType::VARCHAR);
+
+	names.emplace_back("value");
+	return_types.emplace_back(LogicalType::VARCHAR);
+
+	names.emplace_back("scope");
+	return_types.emplace_back(LogicalType::VARCHAR);
+
+	names.emplace_back("scope_entry");
+	return_types.emplace_back(LogicalType::VARCHAR);
+
+	return make_uniq<DuckLakeOptionsData>(catalog);
+}
+
+static string GetOptionDescription(const string &option_name) {
+	for (auto &opt : DUCKLAKE_OPTIONS) {
+		if (StringUtil::CIEquals(opt.name, option_name)) {
+			return opt.description;
+		}
+	}
+	return "Custom configuration option";
+}
+
+unique_ptr<GlobalTableFunctionState> DuckLakeOptionsInit(ClientContext &context, TableFunctionInitInput &input) {
+	auto &bind_data = input.bind_data->Cast<DuckLakeOptionsData>();
+	auto &transaction = DuckLakeTransaction::Get(context, bind_data.catalog);
+	auto &metadata_manager = transaction.GetMetadataManager();
+
+	auto result = make_uniq<DuckLakeOptionsState>();
+	auto metadata = metadata_manager.LoadDuckLake();
+
+	// Global options
+	for (auto &tag : metadata.tags) {
+		DuckLakeOptionInfo option_info;
+		option_info.option_name = tag.key;
+		option_info.value = tag.value;
+		option_info.description = GetOptionDescription(tag.key);
+		option_info.scope = "GLOBAL";
+		result->options.push_back(std::move(option_info));
+	}
+
+	// Schema options
+	for (auto &schema_setting : metadata.schema_settings) {
+		DuckLakeOptionInfo option_info;
+		option_info.option_name = schema_setting.tag.key;
+		option_info.value = schema_setting.tag.value;
+		option_info.description = GetOptionDescription(schema_setting.tag.key);
+		option_info.scope = "SCHEMA";
+		option_info.scope_entry = metadata_manager.GetPathForSchema(schema_setting.schema_id);
```

Comment:
This returns the path - we should return the name instead. You can use `DuckLakeCatalog::GetEntryById` to get the catalog entry which has the name (passing in `transaction.GetSnapshot()`).